backends/ref/ceed-ref-operator.c

```diff
@@ -172,16 +180,21 @@ static inline int CeedOperatorSetupInputs_Ref(CeedInt num_input_fields, CeedQFun
                                               CeedVector in_vec, const bool skip_active, CeedScalar *e_data_full[2 * CEED_FIELD_MAX],
                                               CeedOperator_Ref *impl, CeedRequest *request) {
   for (CeedInt i = 0; i < num_input_fields; i++) {
-    uint64_t            state;
-    CeedEvalMode        eval_mode;
-    CeedVector          vec;
-    CeedElemRestriction elem_rstr;
+    bool         is_active;
+    uint64_t     state;
+    CeedEvalMode eval_mode;
+    CeedVector   vec;
 
     // Get input vector
     CeedCallBackend(CeedOperatorFieldGetVector(op_input_fields[i], &vec));
-    if (vec == CEED_VECTOR_ACTIVE) {
-      if (skip_active) continue;
-      else vec = in_vec;
+    is_active = vec == CEED_VECTOR_ACTIVE;
+    if (is_active) {
+      if (skip_active) {
+        CeedCallBackend(CeedOperatorFieldRestoreVector(op_input_fields[i], &vec));
+        continue;
+      } else {
+        vec = in_vec;
```

Comment:
Continuing this function with `vec = CEED_VECTOR_ACTIVE` seems odd... Like, there are calls to `CeedVectorGetState()`, which is probably valid. But I'm surprised the call to `CeedElemRestrictionApply()` works for `CEED_VECTOR_ACTIVE`